packages/extension-url-mapping/src/setup-app.ts

```diff
@@ -0,0 +1,146 @@
+     * or add things like providers and contexts to be used throughout your app.
+     */
+    extendApp<T extends React.ComponentType<T>>(
+        App: React.ComponentType<T>
+    ): React.ComponentType<T> {
+        // DEBUG
+        console.log(this.getConfig().path)
+        const HOCs = [
+            // Example higher-order component, this can be safely removed.
+            sampleHOC,
+            // Optionally include state for this extension using `withApplicationExtensionStore`
+            (component: React.ComponentType<any>) =>
+                withApplicationExtensionStore(component, {
+                    id: extensionMeta.id,
+                    initializer: sliceInitializer
+                })
+        ]
+
+        return applyHOCs(App, HOCs)
+    }
+
+    /**
+     * TODO: update this comment
+     * This method is used to make changes to the PWA-Kit application routes. If your extension adds a new page to the application
+     * then you can add it to the router here. The routes passed to this method is an accrued list of routes that have been added
+     * from extensions applied before it. It is called during the `getRoutes` phase on both the server and the client.
+     *
+     * NOTE: If you instead want to modify a list of all the routes, refer to the `beforeRouteMatch` below.
+     */
+    async getRoutes({locals}: GetRoutesParams): Promise<RouteProps[]> {
+        const config = this.getConfig()
+
+        if (locals.originalUrl) {
+            const shopperSeo = await getShopperSeoClient(locals, config)
+            const urlMapping = await shopperSeo.getUrlMapping({
+                parameters: {urlSegment: locals.originalUrl}
+            })
+            console.log('--- url mapping', urlMapping)
+        }
+
+        return Promise.resolve([
+            {
+                exact: true,
+                path: this.getConfig().path,
+                component: SamplePage
+            }
+        ])
+    }
+
+    /**
+     * TODO: update comment to make it clearer that beforeRouteMatch is also called on the client side
+     * This method is used on the server during the rendering pipeline. It's provided a list of all the routes that your application
+     * is configured with, including those defined in the base application and those added by all the extensions. You can use this
+     * method to modify these routes in any way you want, but you must return an array of routes as a result.
+     */
+    beforeRouteMatch(allRoutes: RouteProps[]): RouteProps[] {
+        return allRoutes
+    }
+}
+
+export default UrlMapping
+
+const getShopperSeoClient = async (locals: Record<string, any>, config: Config) => {
+    const {
+        commerceAPI,
+        commerceAPIAuth: {propertyNameInLocals: authProperty}
+    } = config
+
+    locals[authProperty] =
+        locals[authProperty] ??
+        new Auth({
+            ...commerceAPI.parameters,
+            proxy: `${getAppOrigin()}${commerceAPI.proxyPath}`,
```

Comment:
We aim to remove this util in favor of `useAppOrigin` to support x-forwarded-header. Since it is a React hook, it can't be used here, we should pass appOrigin as a param to avoid refactoring down the road

Reply:
Thanks Alex.. I'll keep that in mind.